OXRS-SHA-StateMonitor-ESP32-FW.ino

```diff
@@ -126,6 +133,19 @@ void loop()
 {
   // Check our DHCP lease is still ok
   Ethernet.maintain();
+  uint8_t tmp = Ethernet.linkStatus();

```

Comment:
Wondering if we could move this code block into the LCD library - something like `screen.show_ethernet()` - and use this in setup (instead of `show_IP()`).

Just worried this code will have to be copied to every piece of firmware so ideally good to abstract it out into the library if we can? And make it a single call for updating the IP/status in both `setup()` and `loop()`.

This allows scope to add WiFi handling - e.g. `screen.show_wifi()` in the future if needed.

Have you tested it? Does `Ethernet.linkStatus()` work if you pull the plug? I read somewhere this was only supported by some ethernet chips - hoping the Wiznet is one of them as this will be a killer feature!

Reply:
You could then remove the `show_MAC()` call as well, since you could update the MAC via the `show_ethernet()` update, thereby simplifying the library API even further.

Reply:
Yes, I tested the `Ethernet.linkStatus` function and it works very resposive. The W5500 is one of the upported chips